subgraphs/rocket-pool/src/entityUpdates/financialMetrics.ts

```diff
@@ -240,24 +249,40 @@ export function updateProtocolSideRevenueMetrics(
     Address.fromString(PROTOCOL_ID),
     block
   );
+  let additionalRewards = BIGDECIMAL_ZERO;
+
+  const lastPriceUsd = getOrCreateToken(
+    Address.fromString(ETH_ADDRESS),
+    block.number
+  ).lastPriceUSD!;
+
+  const currentEthAmount = bigIntToBigDecimal(pool.inputTokenBalances[1]);
 
-  // Staking rewards revenue is in ETH (rebased in stETH for user), price in ETH
-  const amountUSD = amount;
+  if (amount > currentEthAmount) {
+    additionalRewards = amount.minus(currentEthAmount);
+  }
 
+  const ethAmount = currentEthAmount.plus(additionalRewards);
   // Pool
-  pool.cumulativeProtocolSideRevenueUSD = amountUSD;
+  pool.cumulativeProtocolSideRevenueUSD = ethAmount.times(lastPriceUsd);
   pool.save();
 
   // Pool Daily
   poolMetricsDailySnapshot.cumulativeProtocolSideRevenueUSD =
     pool.cumulativeProtocolSideRevenueUSD;
-  poolMetricsDailySnapshot.dailyProtocolSideRevenueUSD = amountUSD;
+  poolMetricsDailySnapshot.dailyProtocolSideRevenueUSD =
+    poolMetricsDailySnapshot.dailyProtocolSideRevenueUSD.plus(
+      additionalRewards
+    );
   poolMetricsDailySnapshot.save();
 
   // Pool Hourly
   poolMetricsHourlySnapshot.cumulativeProtocolSideRevenueUSD =
     pool.cumulativeProtocolSideRevenueUSD;
-  poolMetricsHourlySnapshot.hourlyProtocolSideRevenueUSD = amountUSD;
+  poolMetricsHourlySnapshot.hourlyProtocolSideRevenueUSD =
+    poolMetricsHourlySnapshot.hourlyProtocolSideRevenueUSD.plus(
+      additionalRewards
```

Comment:
Rewards should be protocolSide or supplySide revenue?

Reply:
rewards likely are protocol side, though I haven't completed those calculations yet. They'll be very similar to eth calculations.